x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteExpiredDataAction.java

```diff
@@ -66,10 +67,12 @@ private void deleteExpiredData(ActionListener<DeleteExpiredDataAction.Response>
 
     private void deleteExpiredData(Iterator<MlDataRemover> mlDataRemoversIterator,
                                    ActionListener<DeleteExpiredDataAction.Response> listener) {
+        Transports.assertNotTransportThread("ML Daily Maintenance");
```

Comment:
This is unnecessary imo. Can we have a comment here why we fork?

Reply:
Yes it was to fail faster during testing/development. I'll remove both `assertNotTransportThread` calls and add a comment